### ISSUE TITLE
Make polyfill optional, at least.

### ISSUE THREAD

**Issue by Axel-Naumann** [doxygen/doxygen#10354]
**Describe the bug**
Requesting mathJax [unconditionally pulls in polyfill.io](https://github.com/doxygen/doxygen/blob/73874def7998419647a6281ec4b39c4261d4d18f/src/htmlgen.cpp#L475). GDPR-wise that's a problem, at least for us. Also, [Browser Support for ES6 (2015)](https://www.w3schools.com/js/js_versions.asp) suggests that this should really not be an issue anymore.

If really need be this could become configurable.

**Expected behavior**
I can use doxygen-generated documentation (including MathJax) without connecting other servers.

**To Reproduce**
See e.g. the network trace of https://root.cern/doc/v628/

**Version**
That's probably not really relevant here - but we use doxygen from Ubuntu 22.04.

**Additional context**
On this topic, [MathJax says](https://docs.mathjax.org/en/latest/output/browser.html)

>  In particular, to allow MathJax version 3 to work with IE11,

[IE11 is dead](https://learn.microsoft.com/en-us/lifecycle/products/internet-explorer-11), please :-)

**Comment by Axel-Naumann** [comment]
I'm happy to provide a PR, ideally just removing the relevant line. Lmk whether that's desirable.

**Comment by albert-github** [comment]
IE11 might look like to be dead but
- the mentioned notice suggests to me that it is still possible to use IE11 to 2029 
- there will be still quite a few systems around that use IE11 (locally and probably also with a local version of MathJax)

Seen this I don't think it is possible yet to remove this line completely, a possibility might be to have (in the resulting HTML file a construct that detects whether or not IE11 is used and when that is the case the part
```
"<script src=\"https://polyfill.io/v3/polyfill.min.js?features=es6\"></script>\n"
```
is enabled.

**Comment by Axel-Naumann** [comment]
Thanks a lot, @albert-github ! The problem I'm trying to solve is that we want our Doxygen doc (which uses MathJax) to be GDPR compliant. For that we must not pull in polyfill, not even if someone uses IE11.

Your comment suggests that this would need to become a configuration option? Maybe as `MATHJAX_VERSION = MathJax_3_nopoly` to not blow up the number of parameters? Or

```
# Dynamically load PolyFill's ES6 support into generated webpages for MathJax v3.
# Requires MATHJAX_VERSION set to MathJax_3.
MATHJAX_POLYFILL = YES
```

All of this is obviously really your call, I'm just suggesting possible approaches hoping that you would agree that GDPR compliance is good to offer, and that you'd tell me how I can contribute to make it happen!

**Comment by albert-github** [comment]
I don't think the polyfill can be removed for IE11 seen the comment you cited from MathJax.
No I'm not suggesting to add an new option for the setting `MATHJAX_VERSION` or a completely new setting `MATHJAX_VERSION` (in case it would be necessary the extension of `MATHJAV_VERSION` would be preferred though).
I think it should be possible in the HTML code to detect the used browser and only when this is `IE11` the `polyfill` should be loaded / enabled.

**Comment by Axel-Naumann** [comment]
I understood your suggestion - apologies for not making that clear. But this means that polyfill will be loaded, even if only in certain cases, and thus websites using Doxygen documentation will not be GDPR compliant.

**Comment by albert-github** [comment]
When I understand it well the `polyfill` has the problem, shouldn't the problem be reported there or with MathJax?

**Comment by doxygen** [comment]
According to [this site](https://www.statista.com/statistics/268254/market-share-of-internet-browsers-worldwide-since-2009/) the worldwide use of IE has dropped to 0.22%. Of those users only a small subset will be using it for doxygen pages, and an even smaller subset will be using Mathjax, so I don't mind to remove the polyfill dependency, and wait for the unlikely bug report that someone finds this a problem.

**Comment by albert-github** [comment]
It doesn't look like much users, but it always depends on what one is measuring, when measuring social networks the results will be different from technical networks. Also there are companies that still use systems that are not (directly) connected to the internet and that run old versions of OS. The later will not likely be using the newest versions of doxygen or update to it so ther risk would be small.

Maybe the suggested `MATHJAX_VERSION = MathJax_3_nopoly` is an easy and good alternative.

**Comment by Axel-Naumann** [comment]
No-network doesn't seem like a scenario where loading polyfill.io would work anyway?

**Comment by albert-github** [comment]
Indeed.
With the MathJax 2 version it was "easy" to create a local, when necessary stripped down, version of MathJax and use this.
With MathJax 3 I never tried a local version but seen the "polyfill" requirement a no-network version won't work at all even when having a local polyfill.js as the polyfill is not configurable.
Regarding the upcoming MathJax 4 I don't know what the possibilities for a no network / local version might be.

**Comment by doxygen** [comment]
I've removed the reference to polyfill.

If someone still needs it do the following:
Generate a custom header
```
doxygen -w html header.html footer.html style.css
```
Edit `header.html` and add `<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>` in front of
`$mathjax`
Use the custom header by setting this in the Doxyfile:
```
HTML_HEADER            = header.html
```

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.10.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
